elastic/beats
# Add cgroup.cpuacct percentages

## What does this PR do?

This PR addresses https://github.com/elastic/beats/issues/23391 and adds percentage values to the `cpuacct` cgroup. Although that change is fairly straightforward, I felt the need to do some refactoring to make it nice--right now, the process metricset tends to divide work between the metricset itself and libbeat in a way that doesn't seem consistent, as libbeat does almost all the heavy lifting for regular process metrics, and then for cgroup data we do all the work in the metricset. 

In order to emulate how process calculates regular cpu percentages, I just had libbeat manage cgroup data the way in manages all the other process data, so we just need to add the bits that deals with cgroup logic, and everything else is handled by the existing process process code. As a result of this, most of the cgroup logic has now been moved to libbeat, since I felt it made organizational sense to not flip-flop between doing the `MapStr` manipulation between two different libraries. 

## Why is it important?

People are asking for the enhancement in https://github.com/elastic/beats/issues/23391

## Checklist


## How to test this PR locally

- Pull down, build
- Run `system/process` on a host with cgroups, and cgroup processes running. 
- Look for the percentages in `system.process.cgroups.cpuacct`, make sure the values are there, and that they make sense.

## Related issues

- https://github.com/elastic/beats/issues/23391
